test/OpenTelemetry.Instrumentation.AspNetCore.Tests/RouteTests/TestApplication/TestApplicationFactory.cs

```diff
@@ -154,4 +162,30 @@ private static WebApplication CreateRazorPagesApplication()
 
         return app;
     }
+
+    private static WebApplication CreateExceptionHandlerApplication()
+    {
+        var builder = WebApplication.CreateBuilder();
+        builder.Logging.ClearProviders();
+
+        var app = builder.Build();
+
+#if NET6_0_OR_GREATER
+        app.UseExceptionHandler(exceptionHandlerApp =>
+        {
+            exceptionHandlerApp.Run(async context =>
+            {
+                context.Response.StatusCode = StatusCodes.Status500InternalServerError;
+                var exceptionHandlerPathFeature = context.Features.Get<IExceptionHandlerPathFeature>();
+                await context.Response.WriteAsync(exceptionHandlerPathFeature?.Error.Message ?? "An exception was thrown.");
+            });
+        });
+#endif
+        app.Urls.Clear();
+        app.Urls.Add("http://[::1]:0");
+
+        app.MapGet("/Exception", (ctx) => throw new ApplicationException());
```

Comment:
> How do you suggest we make the tests pass for 8 until the fix is released?

@cliedeman I suggest we force .NET 8 test to pass as follows, for now. I will open an issue to remind us to update this test once ASP.NET Core 8.0.2 is released.

```suggestion
// TODO: Remove this condition once ASP.NET Core 8.0.2.
// Currently, .NET 8 has a different behavior than .NET 6 and 7.
// This is because ASP.NET Core 8+ has native metric instrumentation.
// When ASP.NET Core 8.0.2 is released then its behavior will align with .NET 6/7.
// See: https://github.com/dotnet/aspnetcore/issues/52648#issuecomment-1853432776
#if !NET8_0_OR_GREATER
        app.MapGet("/Exception", (ctx) => throw new ApplicationException());
#else
        app.MapGet("/Exception", () => Results.Content(content: "Error", contentType: null, contentEncoding: null, statusCode: 500));
#endif
```